pkg/complete/doc.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2012-2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+/*
+	Package complete implements a simple completion package
+	designed to be used in shells and other programs. It currently
+	offers completion functions to implement table-based and
+	file search path based completion. It also offers a multi
+	completion capability so that you can construct completions
+	from other completions.
+
+	Goals:
+	small code base, so it can easily be embedded in firmware
+
+	easily embedded in other programs, like the ip command
+
+	friendly to mixed modes, i.e. if we say
+	ip l
+	and stdin is interactive, it would be nice if ip dropped into
+	a command line prompt and let you use completion to get the rest
+	of the line, instead of printing out a bnf
+
+	The structs should be very light weight and hence cheap to build, use,
+	and throw away. They should NOT have lots of state.
+
+	Rely on the fact that system calls and kernels are fast and cache file system
+	info so you should not. This means that we don't need to put huge effort into building in-memory
+	structs representing file system information. Just ask the kernel.
```

Comment:
it's been, literally, decades since I had a delay that long on a completion that I killed it. Well, until a certain srcfs entered my life.

So, ok, here is my counter argument. It's too soon to do this, but we should do it. But I'd suggest we do it by adding a ContextCompleter, with a New that takes a timeout. The ContextCompleter can wrap any completer and timeout. Then it's up to the caller to decide to use it  or not.